sdks/python/apache_beam/io/source_test_utils.py

```diff
@@ -67,6 +72,40 @@ class ExpectedSplitOutcome(object):
   MUST_BE_CONSISTENT_IF_SUCCEEDS = 3
 
 
+class TestCaseWithTempDirCleanUp(unittest.TestCase):
+  """Base class for TestCases that deals with TempDir clean-up.
+
+  Inherited test cases will call self._new_tempdir() to start a temporary dir
+  which will be deleted at the end of the tests (when tearDown() is called).
+  """
+
+  def setUp(self):
+    self._tempdirs = []
+
+  def tearDown(self):
+    for path in self._tempdirs:
+      if os.path.exists(path):
+        shutil.rmtree(path)
+    self._tempdirs = []
+
+  def _new_tempdir(self):
+    result = tempfile.mkdtemp()
+    self._tempdirs.append(result)
+    return result
+
+  def _create_temp_file(self, name='', suffix='', tmpdir=None, lines=None):
```

Comment:
The tmpdir parameter doesn't seem useful: can't we always use a temporary directory managed by the current instance? (from `_tempdirs`)